doc/src/devdocs/build/windows.md

```diff
@@ -118,11 +118,68 @@ MinGW-w64 compilers available through Cygwin's package manager.
 
 ### Compiling with MinGW/MSYS2
 
-Compiling Julia from source using [MSYS2](https://msys2.github.io) has worked in the past
-but is not actively supported. Pull requests to restore support would be welcome. See a
-[past version of this
-file](https://github.com/JuliaLang/julia/blob/v0.6.0/README.windows.md) for the former
-instructions for compiling using MSYS2.
+> MSYS2 provides a robust MSYS experience.
+
+Note: MSYS2 requires **64 bit** Windows 7 or newer.
+
+ 1. Install and configure [MSYS2](https://www.msys2.org/), Software Distribution
+    and Building Platform for Windows.
+
+    1. Download and run the latest installer for the
+        [64-bit](https://github.com/msys2/msys2-installer/releases/latest) distribution.
+        The installer will have a name like `msys2-x86_64-yyyymmdd.exe`.
+
+    2. Open MSYS2. Update package database and base packages:
+        ```sh
+        pacman -Syu
+        ```
+
+    3. Exit and restart MSYS2, Update the rest of the base packages:
+        ```sh
+        pacman -Syu
+        ```
+
+    3. Then install tools required to build julia:
+        ```sh
+        # tools
+        pacman -S cmake diffutils git m4 make patch tar p7zip curl python
+
+        # For 64 bit Julia, install x86_64
+        pacman -S mingw-w64-x86_64-gcc mingw-w64-x86_64-gcc-fortran
+        # For 32 bit Julia, install i686
+        pacman -S mingw-w64-i686-gcc mingw-w64-i686-gcc-fortran
```

Comment:
Technically we don't need `gfortran` anymore, only if you want to build without BinaryBuilder. :)

Reply:
Done, but it needs to be checked again.

I remember that when compiling with `mingw-w64-clang-x86_64-clang`, it reported an error that `gfortran` was not found.

Reply:
If there are invocations of `$(FC)` that happen automatically, they should have their `stderr` piped to `/dev/null`, [like these examples](https://github.com/JuliaLang/julia/blob/de375ef5167914a5398dd16726e95dfb5e08c392/Make.inc#L1139).  If not, they probably need to be fixed.